src/test/java/org/apache/sysds/test/functions/transform/TransformFederatedEncodeDecodeTest.java

```diff
@@ -0,0 +1,155 @@
+import org.apache.sysds.test.AutomatedTestBase;
+import org.apache.sysds.test.TestConfiguration;
+import org.apache.sysds.test.TestUtils;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class TransformFederatedEncodeDecodeTest extends AutomatedTestBase {
+	private static final String TEST_NAME1 = "TransformFederatedEncodeDecode";
+	private static final String TEST_DIR = "functions/transform/";
+	private static final String TEST_CLASS_DIR = TEST_DIR + TransformFederatedEncodeDecodeTest.class.getSimpleName()
+		+ "/";
+
+	private static final String SPEC = "TransformEncodeDecodeSpec.json";
+
+	private static final int rows = 1234;
+	private static final int cols = 2;
+	private static final double sparsity1 = 0.9;
+	private static final double sparsity2 = 0.1;
+
+	@Override
+	public void setUp() {
+		TestUtils.clearAssertionInformation();
+		addTestConfiguration(TEST_NAME1, new TestConfiguration(TEST_CLASS_DIR, TEST_NAME1, new String[] {"FO"}));
+	}
+
+	@Test
+	public void runTestCSVDenseCP() {
+		runTransformEncodeDecodeTest(false, Types.FileFormat.CSV);
+	}
+
+	@Test
+	public void runTestCSVSparseCP() {
+		runTransformEncodeDecodeTest(true, Types.FileFormat.CSV);
+	}
+
+	@Test
+	public void runTestTextcellDenseCP() {
+		runTransformEncodeDecodeTest(false, Types.FileFormat.TEXT);
+	}
+
+	@Test
+	public void runTestTextcellSparseCP() {
+		runTransformEncodeDecodeTest(true, Types.FileFormat.TEXT);
+	}
+
+	@Test
+	public void runTestBinaryDenseCP() {
+		runTransformEncodeDecodeTest(false, Types.FileFormat.BINARY);
+	}
+
+	@Test
+	public void runTestBinarySparseCP() {
+		runTransformEncodeDecodeTest(true, Types.FileFormat.BINARY);
+	}
+
+	private void runTransformEncodeDecodeTest(boolean sparse, Types.FileFormat format) {
+		ExecMode platformOld = rtplatform;
+		rtplatform = ExecMode.SINGLE_NODE;
+
+		Thread t1 = null, t2 = null;
+		try {
+			getAndLoadTestConfiguration(TEST_NAME1);
+
+			int port1 = getRandomAvailablePort();
+			t1 = startLocalFedWorker(port1);
+			int port2 = getRandomAvailablePort();
+			t2 = startLocalFedWorker(port2);
+
+			// schema
+			Types.ValueType[] schema = new Types.ValueType[cols / 2];
+			Arrays.fill(schema, Types.ValueType.FP64);
+			// generate and write input data
+			// A is the data that will be aggregated and not recoded
+			double[][] A = TestUtils.round(getRandomMatrix(rows, cols / 2, 1, 15, sparse ? sparsity2 : sparsity1, 7));
+			writeInputFrameWithMTD("A", A, false, schema, format);
+
+			// B will be recoded and will be the column that will be grouped by
+			Arrays.fill(schema, Types.ValueType.STRING);
+			// we set sparsity to 1.0 to ensure all the string labels exist
+			double[][] B = TestUtils.round(getRandomMatrix(rows, cols / 2, 1, 15, 1.0, 8));
+			writeInputFrameWithMTD("B", B, false, schema, format);
+
+			fullDMLScriptName = SCRIPT_DIR + TEST_DIR + TEST_NAME1 + ".dml";
+
+			programArgs = new String[] {"-explain", "-args", TestUtils.federatedAddress("localhost", port1, input("A")),
```

Comment:
If you can live without calling explain in tests it would be great. since it has a tendency to write alittle to much to the terminal, that in turn crash our test suite on workflows.